src/env/node/git/localGitProvider.ts

```diff
@@ -2001,6 +2005,16 @@ export class LocalGitProvider implements GitProvider, Disposable {
 							upstream: branch?.upstream?.name,
 						};
 						refHeads.push(refHead);
+						if (branch?.upstream?.name != null) {
+							// Add the branch name (tip) to the upstream name entry in the downstreams map
+							let downstreams = downstreamMap.get(branch.upstream.name);
+							if (downstreams == null) {
+								downstreams = [];
+							}
+
+							downstreams.push(tip);
+							downstreamMap.set(branch.upstream.name, downstreams);
```

Comment:
Can move this up, as you only need to re-set the array in the map when its `undefined`

Reply:
Done in commit: [Optimizations](https://github.com/gitkraken/vscode-gitlens/pull/2552/commits/822bbc116c4740fb919c2a1af03d019b4a56e701)